### DISCUSSION TITLE
Removing particles over land, and trajectories across land

### DISCUSSION THREAD

**Discussion by claudioiturra** [Parcels-code/Parcels#1532] [General]
Hello, I will appreciate any help on how to removing particles over land, and trajectories across land.

![Screenshot from 2024-04-05 19-00-42](https://github.com/OceanParcels/parcels/assets/61473481/9c41c7c6-580b-4c7d-ae24-d3d2d83360a4)

**Comment by michaeldenes** [comment]
HI @claudioiturra, I assume you're using the cartopy coastlines approach to plot the coast/land? In nearly all cases, the cartopy coastline does not match the model land cells, and you'll often have ocean grid cells in cartopy 'land' locations, and land grid cells in cartopy 'ocean' locations. Could you make a similar plot, using the model mesh to show the coast, rather than with cartopy?

Regarding the stuck particles, you can see the approach in one of the tutorials here: https://docs.oceanparcels.org/en/latest/examples/documentation_stuck_particles.html
Everything will depend on the particular model setup you have, whether it's an A-grid, B-grid, or a C-grid.

Cheers,
Michael

**Reply by claudioiturra** [reply]
Hi @michaeldenes, Sorry, but how do I make the same plot using the model mesh?, is the mesh of Hycom?, I thought that the model can evaluate and adjust the direction of  the current velocity in areas near shore or bays?, right?

**Reply by michaeldenes** [reply]
Hi @claudioiturra, the linked tutorial above, along with https://docs.oceanparcels.org/en/latest/examples/tutorial_nemo_3D.html have some good examples of plotting the velocity fields (which should show you where the model boundaries are. I'm not sure what model you're using exactly, but if it's HYCOM, you can find a mesh file for it. I'm more familiar with NEMO and MOM model outputs, but parcels essentially solves the advection equation using whatever velocity fields you pass it.

The linked tutorial in my above post highlights some of the issues with A- and B-grids, where you can get artificial beaching due to the way boundary conditions are handled (an issue of using such grids).
